src/ProcessOutput.php

```diff
@@ -50,17 +49,17 @@ public function __construct($countOfFiles)
     public function process($statusCode)
     {
         $symbol = sprintf(self::$statusMap[$statusCode]['format'], self::$statusMap[$statusCode]['symbol']);
-        $this->progressLine[$this->currentPosition] = $symbol;
-        $this->currentPosition++;
+        $this->progressLine[] = $symbol;
+        $this->currentPosition = count($this->progressLine);
 
         $percentOfComplete = $this->currentPosition * 100 / $this->countOfFiles;
 
-        end($this->progressLine);
-        $lastKey = key($this->progressLine);
-        $this->progressLine[$lastKey] = sprintf('%d/%d (%d%%)', $this->currentPosition, $this->countOfFiles, $percentOfComplete);
-        $progressLine = implode('', $this->progressLine);
+        $progress = sprintf('%d/%d (%3d%%)', $this->currentPosition, $this->countOfFiles, $percentOfComplete);
+        $wrappedProgressLines = array_chunk($this->progressLine, $this->maxColumnWidth);
+        $wrapLine = $this->currentPosition % $this->maxColumnWidth == 0 ? PHP_EOL : '';
+        $progressLineFormat = "\r%-" . $this->maxColumnWidth . 's%+' . $this->progressLength . 's%s';
```

Comment:
Can it be simplified? I don't understand this code

Reply:
it will reprint the progress line till there is space else the progress will continue on the next line and the previous line will be persisted in console.

Reply:
I thought 60. line...others is ok but 60. line is too complicated for me

Reply:
i needed to use this knowledge http://php.net/manual/en/function.sprintf.php#refsect1-function.sprintf-parameters for some characters but generally 
`\r` - will return text cursor position to the beggining of the line
`-` will make string left justified
`maxcolumnwidth` - size of the allocated space for the string
`+` - will make string right justified (should be by default but who knows)
`progressLength` - is also for space allocation for the string of second input value
`%s` - last string is allocated for wrapping of line based on wrapping method `array_chunk`